sentry-ruby/lib/sentry/transaction.rb

```diff
@@ -64,7 +64,7 @@ def deep_dup
     end
 
     def set_initial_sample_decision(sampling_context:)
-      unless configuration.tracing_enabled?
+      unless configuration.tracing_enabled? && configuration.sending_allowed?
         @sampled = false
         return
       end
```

Comment:
Tracing being enabled or not should probably not interfere with the sampling decision.

We don't have an equivalent of `enabled_environments` and `sending_allowed` in other SDKs, so I'm thinking here if it makes sense to have `tracing_enabled == true` and `sending_allowed == false`, or if `sending_allowed == false` should imply `tracing_enabled = false`.

Reply:
I didn't expect the later option to work without breaking tons of test cases, but I was wrong. So I've rewritten the commits and the description. Thanks for the suggestion 👍